src/main/java/org/jenkinsci/plugins/cloudstats/CloudStatistics.java

```diff
@@ -475,6 +524,17 @@ protected void doRun() throws Exception {
         return ((TrackedItem) plannedNode).getId();
     }
 
+    private static @CheckForNull ProvisioningActivity.Id getIdFor(Node node) {
+        if (node instanceof Jenkins) return null;
+
+        if (!(node instanceof TrackedItem)) {
+            LOGGER.info("No support for cloud-stats-plugin by " + node.getClass());
```

Comment:
I guess so. We should do it once per type. Though, this behavior is not related to this PR, I presume that is how you found that out.